spring-integration-file/src/main/java/org/springframework/integration/file/filters/FileListFilter.java

```diff
@@ -38,4 +39,26 @@
 	 */
 	List<F> filterFiles(F[] files);
 
+	/**
+	 * Filter a single file.
+	 * @param file the file.
+	 * @return true if the file passes the filter, false to filter.
+	 * @since 5.2
+	 */
+	default boolean accept(F file) {
+		@SuppressWarnings("unchecked")
+		F[] fileArray = (F[]) Array.newInstance(file.getClass(), 1);
+		fileArray[0] = file;
```

Comment:
How about `F[] fileArray = (F[]) Stream.of(file).toArray();` instead?

Reply:
```
Exception in thread "main" java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [Ljava.lang.String;
```

Reply:
Hm. So, the `Array.newInstance` is still need to be used, but `Stream` can remain yet 😀

Reply:
Actually, it can be replaced with an `UnsupportedOperationException`, or `IllegalStateException`. I added this before adding the `supportsSingleFileFiltering`; any filter that returns true there must implement this method.

Reply:
Ha! Good catch!
Yes, let's go with the `UnsupportedOperationException` since `supportsSingleFileFiltering()` returns `false`!